File: ABC_Chapter3_Exercise3/app/src/main/java/com/example/cyu/abc_chapter3_exercise3/MainActivity.java
package com.example.cyu.abc_chapter3_exercise3;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;

import java.text.NumberFormat;

public class MainActivity extends AppCompatActivity {

    double gallonsPerSqFt = 250;
    double numberOfGallons;
    double totalCost;
    String groupChoice;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final EditText gallons = findViewById(R.id.txtGallons);
        final Spinner group = findViewById(R.id.txtGroup);

        Button cost = findViewById(R.id.btnCost);
        cost.setOnClickListener(new View.OnClickListener() {
            final TextView result = findViewById(R.id.txtResult);

            @Override
            public void onClick(View view) {
                numberOfGallons = Integer.parseInt(gallons.getText().toString());
                totalCost = gallonsPerSqFt * numberOfGallons;
                NumberFormat currency = NumberFormat.getCurrencyInstance();
                groupChoice = group.getSelectedItem().toString();
                result.setText("Cost for " + groupChoice + " is " + currency.format(totalCost));
            }
        });
    }
}

File: ABC_Chapter3_Exercise3/app/src/main/java/com/example/cyu/abc_chapter3_exercise3/MainActivity.kt
package com.example.cyu.abc_chapter3_exercise3

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView

import java.text.NumberFormat

class MainActivity : AppCompatActivity() {

    internal var gallonsPerSqFt = 250.0
    internal var numberOfGallons: Double = 0.toDouble()
    internal var totalCost: Double = 0.toDouble()
    internal lateinit var groupChoice: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val gallons = findViewById<EditText>(R.id.txtGallons)
        val group = findViewById<Spinner>(R.id.txtGroup)

        val cost = findViewById<Button>(R.id.btnCost)
        cost.setOnClickListener(object : View.OnClickListener {
            val result = findViewById<TextView>(R.id.txtResult)

            override fun onClick(view: View) {
                numberOfGallons = Integer.parseInt(gallons.text.toString()).toDouble()
                totalCost = gallonsPerSqFt * numberOfGallons
                val currency = NumberFormat.getCurrencyInstance()
                groupChoice = group.selectedItem.toString()
                result.text = "Cost for " + groupChoice + " is " + currency.format(totalCost)
            }
        })
    }
}
